plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -271,8 +253,301 @@ type Query {
+        run => ({
+          ...run,
+          status: "FAILED" as MetronomeClient.JobStatus
+        })
+      );
+
+      return typeResolvers.JobRunConnection([
+        ...job.activeRuns,
+        ...successfulFinishedRunsWithStatus,
+        ...failedFinishedRunsWithStatus
+      ]);
+    },
+    json(job: MetronomeClient.JobDetailResponse): string {
+      return JSON.stringify(cleanJobJSON(job));
+    },
+    labels(job: MetronomeClient.JobDetailResponse): Label[] {
+      return Object.entries(job.labels).map(([key, value]) => ({ key, value }));
+    },
+    lastRunStatus(job: MetronomeClient.JobDetailResponse): JobRunStatusSummary {
+      return typeResolvers.JobRunStatusSummary(job);
+    },
+    lastRunsSummary(job: MetronomeClient.JobDetailResponse): JobHistorySummary {
+      return typeResolvers.JobHistorySummary(job.history);
+    },
+    name(job: MetronomeClient.JobDetailResponse): string {
+      return job.id.split(".").pop() || "";
+    },
+    schedules(job: MetronomeClient.JobDetailResponse): ScheduleConnection {
+      return typeResolvers.ScheduleConnection(job.schedules);
+    },
+    scheduleStatus(job: MetronomeClient.JobDetailResponse): JobStatus {
+      const activeRuns = new JobRunList({ items: job.activeRuns });
+      const activeRunsLength = activeRuns.getItems().length;
+      const scheduleLength = job.schedules.length;
+
+      if (activeRunsLength > 0) {
+        const longestRunningActiveRun = activeRuns.getLongestRunningActiveRun();
+
+        return longestRunningActiveRun.status;
+      }
+
+      if (scheduleLength > 0) {
+        const schedule = job.schedules[0];
+
+        if (schedule != null && schedule.enabled) {
+          return "SCHEDULED";
+        }
+      }
+
+      if (scheduleLength === 0 && activeRunsLength === 0) {
+        return "UNSCHEDULED";
+      }
+
+      return "COMPLETED";
+    }
+  },
+  JobRun: {
+    jobID(
+      run: MetronomeClient.ActiveJobRun | MetronomeClient.JobHistoryRun
+    ): string {
+      if (isActiveRun(run)) {
+        return run.jobId;
+      }
+
+      if (isHistoryRun(run)) {
+        return run.id;
+      }
+
+      // Only for typescript, should be impossible
+      return "";
+    },
+    dateFinished(
+      run: MetronomeClient.ActiveJobRun | MetronomeClient.JobHistoryRun
+    ): number | null {
+      if (isActiveRun(run) && run.completedAt) {
+        return DateUtil.strToMs(run.completedAt);
+      }
+      return null;
+    },
+    tasks(
+      run: MetronomeClient.ActiveJobRun | MetronomeClient.JobHistoryRun
+    ): JobTaskConnection {
+      return typeResolvers.JobTaskConnection(isActiveRun(run) ? run.tasks : []);
+    }
+  },
+  JobRunConnection: {
+    longestRunningActiveRun(
+      runs: Array<MetronomeClient.ActiveJobRun | JobHistoryRunWithStatus>
+    ): JobRun | null {
+      if (!runs.length) {
+        return null;
+      }
+
+      const sortedRuns = runs.sort((a, b) => {
+        if (a.createdAt == null && b.createdAt == null) {
```

Comment:
Should we avoid the type coercion here? (===) or we are also guarding against undefined?

Reply:
Also here if we find the base value that would make `DateUtil.strToMs(baseValue) === 0`, we could write:

```js
return DateUtil.strToMs(a.createdAt || baseValue) - DateUtil.strToMs(b.createdAt || baseValue));
```